openedx_caliper_tracking/tests/test_caliper_kafka.py

```diff
@@ -0,0 +1,157 @@
+import json
+import os
+
+import mock
+from django.conf import settings
+from django.test import TestCase, override_settings
+from kafka.errors import KafkaError
+
+from openedx_caliper_tracking.processor import CaliperProcessor
+from openedx_caliper_tracking.tasks import _host_not_found
+
+TEST_DIR_PATH = os.path.dirname(os.path.abspath(__file__))
+
+
+class CaliperKafkaTestCase(TestCase):
+
+    def setUp(self):
+        input_file = '{}/current/{}'.format(
+            TEST_DIR_PATH,
+            'book.json'
+        )
+        with open(input_file) as current:
+            self.event = json.loads(current.read())
+
+    @mock.patch(
+        'openedx_caliper_tracking.tasks.KafkaProducer',
+        autospec=True
+    )
+    @override_settings(
+        CALIPER_KAFKA_SETTINGS={
+            'END_POINT': '192.168.10.5:9092',
+            'TOPIC_NAME': 'dummy topic',
+            'ERROR_REPORT_EMAIL': 'dummy@example.com',
+            'MAXIMUM_RETRIES': 3
+        },
+        FEATURES={'ENABLE_KAFKA_FOR_CALIPER': True}
+    )
+    def test_caliper_event_is_delivered_to_kafka_without_error(self, producer_mock):
+        """
+        Test that  caliper event is delivered with all required settings given.
+        """
+        CaliperProcessor().__call__(self.event)
+        self.assertTrue(producer_mock.called)
+
+    @mock.patch(
+        'openedx_caliper_tracking.tasks.send_notification',
+        autospec=True
+    )
+    @mock.patch(
+        'openedx_caliper_tracking.tasks.KafkaProducer',
+    )
+    @override_settings(
+        CALIPER_KAFKA_SETTINGS={
+            'END_POINT': '192.168.10.5:9092',
```

Comment:
Why are we using this specific IP for the test ?

Reply:
it's just a dummy one.

Reply:
Could we use a simple one? like `localhost` to avoid confusion.